autopilot/contractor.go

```diff
@@ -588,22 +613,22 @@ func (c *contractor) runContractChecks(ctx context.Context, w Worker, contracts
 				"refresh", refresh,
 				"renew", renew,
 			)
+		} else {
+			// a usable contract will be kept even if it also needs to be
+			// renewed/refreshed.
+			toKeep = append(toKeep, fcid)
```

Comment:
I'm not sure whether adding `renew` and `refresh` contracts to `toKeep` here is the easiest and most readable way to accomplish what you are trying to do. It's because I think by doing this you complicate keeping it an actual set. Why don't we keep `runContractChecks` the way it is right now, but handle `toRenew` and `toRefresh` differently? 

(because you have to remove `renewedFCID` from the updated set after renewing it...)